rpc/transaction.go

```diff
@@ -224,8 +224,9 @@ type Transaction struct {
 }
 
 type TransactionStatus struct {
-	Finality  TxnStatus          `json:"finality_status"`
-	Execution TxnExecutionStatus `json:"execution_status,omitempty"`
+	Finality      TxnStatus          `json:"finality_status"`
+	Execution     TxnExecutionStatus `json:"execution_status,omitempty"`
+	FailureReason string             `json:"failure_reason,omitempty"`
```

Comment:
is it true that we get this value by unmarshaling of some response or we should set it explicitly somewhere? I found `adaptTransactionStatus` function that set 2 out of 3 fields

Reply:
Good point, the spec requires that [getTransactionStatus](https://github.com/starkware-libs/starknet-specs/blob/v0.8.0-rc0/api/starknet_api_openrpc.json#L225) should return the failure reason. Should I update this PR to reflect those changes, or would it be cleaner to do in a separate PR?

Reply:
The `revert_error` is used for both rejected and reverted transactions

Reply:
Thanks, it's strange that we hasn't used `txStatus.RevertError` before